frontend/src/components/AccountsList.js

```diff
@@ -3,71 +3,94 @@ import {getCurrencyNameById, isFiat} from "../utils/CryptoUtils";
-                                    ca_number: account.ca_number,
-                                    fk_cc_id: account.fk_cc_id
-                                }
-                            })
-                        }}>Вывести</Button>
-                    </div>
-                } else {
-                    buttons = <div>
-                        <Button
-                            onClick={() => {
-                                navigate("/buy", {
-                                    state: {
-                                        ca_id: account.ca_id,
-                                        ca_number: account.ca_number,
-                                        fk_cc_id: account.fk_cc_id,
-                                        fk_user_id: account.fk_user_id
-                                    }
-                                });
-                            }}
-                        >Купить
-                        </Button>
-                    </div>
+        !accounts ? "" :
+            accounts.map(account => {
+                    var buttons;
+                    if (isFiat(account.fk_cc_id)) {
+                        buttons = <div>
+                            <Button
+                                onClick={() => {
+                                    navigate("/topup", {
+                                        state: {
+                                            ca_id: account.ca_id,
+                                            ca_number: account.ca_number,
+                                            fk_cc_id: account.fk_cc_id
+                                        }
+                                    })
+                                }}>Пополнить</Button>
+                            <Button
+                                onClick={() => {
+                                    navigate("/withdrawal", {
+                                        state: {
+                                            ca_id: account.ca_id,
+                                            ca_number: account.ca_number,
+                                            fk_cc_id: account.fk_cc_id
+                                        }
+                                    })
+                                }}>Вывести</Button>
+                            <Button
+                                onClick={() => {
+                                    navigate("/buy", {
+                                        state: {
+                                            ca_id: account.ca_id,
+                                            ca_number: account.ca_number,
+                                            fk_cc_id: account.fk_cc_id,
+                                            fk_user_id: account.fk_user_id
+                                        }
+                                    });
+                                }}
+                            >Купить
+                            </Button>
```

Comment:
Create component, like WithdrawalButton(account). Just extraction and copy&paste